### ISSUE TITLE
SideNavMenuItem doesn't pass IBM's DAP role hierarchy test

### ISSUE THREAD

**Issue by fsegu002** [carbon-design-system/carbon#4944]
## What package(s) are you using?

- [ ] `carbon-components@10.7.4`
- [x] `carbon-components-react@7.7.4` 

## Detailed description
After running DAP tool I get the following violation. 
![image](https://user-images.githubusercontent.com/3401787/71679953-d445c100-2d56-11ea-8544-5e4295c95506.png)

After taking a closer look at the source code I can see that the `<li>` element has `role="none"` and the `<a>` element has `role="menuitem"` which invokes the violation.

> Is this issue related to a specific component?
Yes. `<SideNavMenuItem>`

> What browser are you working in?
Chrome

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
No

## Steps to reproduce the issue
https://codesandbox.io/s/wizardly-kilby-flc8r

1.  Click open in new window
2.  Open Developer Tools
3.  Under "Elements" tab look for "DAP" subtab
4.  Press play button to run tool

## Additional information

- The error is coming from [this](https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/UIShell/SideNavMenuItem.js#L27) line

**Comment by abbeyhrt** [comment]
Hi @fsegu002! Thank you for taking the time to make this issue! We already have an issue (#3572) documenting this problem and a PR (#4834)  in the works to address it. I'm going to close this issue just so we don't have any duplicates floating around.

**Comment by electrostaticfleece** [comment]
@abbeyhrt @joshblack @asudoh 

Can this be re-opened? It appears as though #4834 addressed issues with the `HeaderMenu` and the `HeaderMenuItem`, but this appears to be a separate problem with `SideNavMenuItem`. 

Running DAP on the Header Base with Side Nav variation of the UI shell with one of the menus open produces the following results for me:

<img width="1490" alt="Screen Shot 2020-07-13 at 3 17 04 PM" src="https://user-images.githubusercontent.com/16194732/87350090-f8ac0280-c51c-11ea-96a3-d3677057c66e.png">

I believe the above issue still exists and should be corrected. Please let me know if this is being tracked elsewhere. Currently, working on a few a11y issues for IBM Cloud Docs and this DAP violation was reported recently.

**Comment by carmacleod** [comment]
To fix completely, please do the following:
- delete `role="none"` from the `<li>` element on [SideNavMenuItem.js line 26](https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/UIShell/SideNavMenuItem.js#L26)
  - because `role="none"` on a list item prevents it from being counted as a member of the list (i.e. screen readers like to say "list with 5 items", "2 of 5", etc)
- delete `role="menuitem"` from the `<a>`element on [SideNavMenuItem.js line 27](https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/UIShell/SideNavMenuItem.js#L27)
  - because menu items need to be in a menu in order to function properly
- remove `aria-haspopup="true"` from the `<button>`
  - because this turns a button into a "menu button", which would need to have a menu and menuitems, plus proper menu keyboard handling (like down arrow opens menu, which I think you do not want here because down arrows go to next link)
  - definitely keep the `aria-expanded="true"/"false"` on the button, so that screen reader users know that activating the button will give them more stuff (links, in this case, because they know they're in a nav)
- fix up any of your css that was relying on the `aria-haspopup` attribute being there

i.e. the idea is that you need to choose whether to go all-in with menus, including all the markup and all the keyboard handling and only a single tab stop for the whole menu..... OR you need to take out the old menu-related markup (which is the suggested fix, above) and then you can use the keyboard handling that you currently have, where up/down and tab key go to the next link, and enter/space on button expands to show more links. Hope this makes sense.
